scripts/solve_network.py

```diff
@@ -155,10 +155,21 @@ def add_chp_constraints(n):
 
         define_constraints(n, lhs, "<=", 0, 'chplink', 'top_iso_fuel_line')
 
+def add_land_use_constraint(n):
+    for carrier in ['solar', 'onwind', 'offwind-ac', 'offwind-dc']:
+        gens = list(n.generators.index[n.generators.carrier==carrier])
+        gens_0=[gen for gen in gens if gen[-4:] not in snakemake.config['scenario']['planning_horizons']]        
+        #total capacity installed in a country <= potential per country
+        lhs=linexpr((1, np.array([n.generators.loc[[gen for gen in gens if country[0:3] in gen],"p_nom"].sum() for country in gens_0])),
+                    (-1, n.generators.loc[gens_0,'p_nom_max'].values))
+        
+        define_constraints(n, lhs, "<=", 0, "land use", carrier)
+
 def extra_functionality(n, snapshots):
     #add_opts_constraints(n, opts)
     #add_eps_storage_constraint(n)
     add_chp_constraints(n)
+    add_land_use_constraint(n)
     add_battery_constraints(n)
```

Comment:
Won't add_land_use break for the overnight foresight case? Maybe make it conditional on config["foresight"]?

Reply:
Yes, you are right. Implemented in https://github.com/PyPSA/pypsa-eur-sec/pull/63/commits/dedabe695ed7da305e05d86c05a5938414b18040